okio/src/commonMain/kotlin/okio/FileSystem.kt

```diff
@@ -153,6 +153,16 @@ abstract class FileSystem {
   @Throws(IOException::class)
   abstract fun source(file: Path): Source
 
+  /**
+   * Creates a source to read [file], executes [readerAction] to read it, and then closes the
+   * source. This is a compact way to read the contents of a file.
+   */
+  inline fun <T> read(file: Path, readerAction: BufferedSource.() -> T): T {
```

Comment:
I don't think I like these names. Maybe `useSource` and `useSink`? `withSource` and `withSink`?